portfolio/src/main/webapp/script.js

```diff
@@ -65,7 +65,15 @@ function grow() {
 }
 
 function getGreeting() {
-  fetch('/data').then(response => response.text()).then(greeting => {
-    document.getElementById('greeting-container').innerHTML = greeting;
+  fetch('/data').then(response => response.json()).then(greetingsJson => {
+    console.log(greetingsJson);
+    const greetingsElement = document.getElementById('greetings-container');
+    var greeting;
+    var greetingElement;
```

Comment:
https://google.github.io/styleguide/jsguide.html#features-use-const-and-let

https://www.freecodecamp.org/news/var-let-and-const-whats-the-difference/

Since greetingElement is a variable that you are only using inside the for block then its ok to use a let inside so it stays only inside the for block scope.

For the greeting variable declaration, you can do `for (const greeting of greetingsJson)`

Reply:
I was always fuzzy about when to use var/let/const. Thanks for pointing this out!